cosmos/job/drm/drm_k8s_jobs.py

```diff
@@ -0,0 +1,137 @@
+import dateutil.parser
+import json
+import subprocess as sp
+
+from sqlalchemy import inspect as sqlalchemy_inspect
+
+from cosmos.api import TaskStatus
+from cosmos.job.drm.DRM_Base import DRM
+from cosmos.models.Task import Task
+
+
+class DRM_K8S_Jobs(DRM):  # noqa
+
+    """Uses Kubernetes jobs as a method of dispatching tasks. The job manager must be
+        configured to use a specific Docker image to use this DRM.
+    """
+
+    name = 'k8s-jobs'
+    always_cleanup = True
+    required_drm_options = {'image'}
+    optional_drm_options = {'file', 'time', 'name', 'container_name', 'cpu', 'memory', 'disk',
+                            'cpu-limit', 'memory-limit', 'disk-limit', 'time', 'persistent-disk-name',
+                            'volume-name', 'mount-path', 'preemptible'}
+    drm_options_to_task_properties = {
```

Comment:
time_req/cpu_req/mem_req are normally passed through get_submit_args
see cosmos' default_get_submit_args
and Balrog's get_submit_args

Reply:
In the context of Kubernetes, the abstraction of DRMs between the application and the library cosmosI think the underlying application should specify the requirements declaratively, and then cosmos is responsible for the underlying translation between the req spec and invocating the DRM in such a way that it complies. I'll modify the code to still call it and append whatever it returns to the cmd string with the expectation that it will probably never return anything but an empty string. A user will never have to override it in any practical scenario but technically if they had a really weird reason, we are honoring the contract.